### ISSUE TITLE
Filter by a nested "keys"?

### ISSUE THREAD

**Issue by Whyser** [JoeRoddy/firestation#10]
Hey, thanks for this great visualization tool.

Sorry if this is a stupid question but, I'd like to query by a unknown "key". I have the following structure (not my actual use-case).
```
{
   'users':
   {
      '7894hut988g3':
      {
         'username': 'jack'
      }
      'jhbsfbhsfa989':
      {
         'username': 'jimbo'
         'friends':
         {
             'bfhbsjfs87878':
             {
                'username':'jack'
             }
         }
      }
   }
}
```

And I would like to do something like:
`select * from users.$.friends.$.username = 'jack'`
Where $ is, as you might understand arbitrary since I don't know it's value/key. Right now I can only get:
`select * from users where friends != null to work`

**Comment by JoeRoddy** [comment]
I don't think this is supported at the moment, but I can't totally remember.  Sorry, I haven't worked on this project in a while.

You could try something like:
`select * from (select friends from users) where username = 'jack';`

I **think** I have that type of nested query working on the [projectRevamp branch](https://github.com/JoeRoddy/firestation/issues/5).  You could follow the instructions to launch that locally from that link and give it a shot.  Good luck!

**Comment by Whyser** [comment]
Thanks Joe, I will try it out later! In-case you know off the bat, do you _**think**_ this is available for Fire_base_ (notices that the new topic talks alot about Fire_store_?
